tests/unit/utils/test_objectives.py

```diff
@@ -79,6 +81,104 @@ def test_logarithmic_goldstein_price_no_function_values_are_less_than_global_min
     )
 
 
+@pytest.mark.parametrize(
+    "test_x, expected",
+    [
+        (
+            tf.constant([[0.0, 0.0]]),
+            tf.convert_to_tensor([[1.0 - tf.math.exp(-1.0), 1.0 - tf.math.exp(-1.0)]]),
```

Comment:
why do we have all the tf.convert_to_tensor ?

Reply:
CMIIW here, it seems `tf.constant` does not support list of `Tensor`s, which is generated because of `tf.math.exp`, so the `convert_to_tensor` is used instead for converting compatibility on tensors.
The line 89 is written as an expression (thus having `tf.math.exp`) not exact number is more for interpretability, as it is the math definition of the synthetic function.

Reply:
you can write this
```python
[[1.0, 1.0]] - tf.exp(-1.0)
```

Reply:
I think you could just put the exact numbers in (like you do in the next test). Means that this will be much neater and not take up 25 lines of code.

Reply:
Changed to exact numbers for neater expression.